localstack/dns/server.py

```diff
@@ -247,6 +247,7 @@ def handle(self, *args, **kwargs):
 
 NAME_PATTERNS_POINTING_TO_LOCALSTACK = [
     f".*{LOCALHOST_HOSTNAME}",
+    f".*{config.LOCALSTACK_HOST.host}",
```

Comment:
Question: if `LOCALSTACK_HOST` is not overridden, its default value is `localhost.localstack.cloud` so we add the same name twice. I don't think this is a _problem_, but it is not very clean. Perhaps `Resolver.zones` should be a `dict[str, set[NameRecord]]`?

Reply:
It should not be a problem, but we might return multiple records for `localhost.localstack.cloud` then - not really clean.

Can we add a simple guard to avoid adding it if it is `LOCALHOST_HOSTNAME`?

Reply:
Sure but a set is more elegant! That sounds like a reasonable first step anyway

Reply:
My issue with the set is the following: We expect on delete a value error. If two different users of the DNS server add a similar entry, and both delete it afterwards, we get one value error - although there were two creates and two deletes.
These cases would need to be documented clearly, so users could ignore the value error on delete.